synapse/util/metrics.py

```diff
@@ -108,7 +108,12 @@ class Measure:
     def __init__(self, clock, name):
         self.clock = clock
         self.name = name
-        parent_context = current_context()
+        curr_context = current_context()
+        if not curr_context:
+            parent_context = None
+        else:
+            assert isinstance(curr_context, LoggingContext)
+            parent_context = curr_context
```

Comment:
This is an actual behavior change -- are we expecting to measure things started by the sentinel context? Should we add a warning here about this?

Reply:
I don't think we're expecting it, but I certainly can't guarantee that we're not doing it. A warning might be a better first step.

Reply:
Well this assertion should never fire since `curr_context` will be the sentinel (in which case it is false-y) or a `LoggingContext`, the assertion is just there for mypy. I can add a warning though.